pkg/operator/operator.go

```
@@ -321,6 +305,41 @@ func (o *Operator) Run(ctx context.Context, ors ...metav1.OwnerReference) error
 	return o.manager.Start(ctx)
 }
 
+// ensureCerts writes the cert/key files to the specified directory.
+// If cert/key are not avalilable, generate them.
+func (o *Operator) ensureCerts(ctx context.Context, dir string) ([]byte, error) {
+	var (
+		crt, key []byte
+		err      error
+	)
+	if len(o.opts.Key) > 0 && len(o.opts.Cert) > 0 {
```

We should throw an error if only one is set.